api/citations/utils.py

```diff
@@ -42,6 +47,36 @@ def preprint_csl(preprint, node):
     return csl
 
 
+def process_name(node, user):
+    if user.family_name and user.given_name:
+        """If the user has a family and given name, use those"""
+        return {
+            'family_name': user.family_name,
+            'suffix': user.suffix,
+            'given_name': user.given_name,
+            'middle_names': user.middle_names,
+        }
+    else:
+
+        if user.is_registered or user.is_disabled:
+            name = user.fullname
+        else:
+            name = user.get_unclaimed_record(node._id)['name']
+
+        """ If the user doesn't autofill his family and given name """
+        parsed = utils.impute_names(name)
+        given_name = parsed['given']
```

Comment:
It doesn't seem necessary to assign each key to a variable, especially since each variable is only used once. 
```
return {
    'family_name': parsed['family'],
    'suffix': parsed['suffix'],
    'given_name': parsed['given'],
    'middle_names': parsed['middle']
}
``` 
is just as readable to me.